src/uct/tcp/tcp_ep.c

```diff
@@ -258,6 +240,13 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_ep_t, uct_tcp_iface_t *iface, int fd,
     ucs_queue_head_init(&self->pending_q);
     ucs_queue_head_init(&self->put_comp_q);
 
+    if (dest_addr != NULL) {
+        memcpy(&self->peer_addr[0], dest_addr, iface->config.sockaddr_len);
+    } else {
+        memset(&self->peer_addr[0], 0, iface->config.sockaddr_len);
```

Comment:
why needed to reset to 0?

Reply:
unresolved

Reply:
@yosefe  it was there earlier also (called uct_tcp_ep_addr_cleanup() if addr is NULL).  we used "peer_addr" for hashing for conn_match. do you see any problem if not?

Reply:
@dmitrygx WDYT?

Reply:
> @dmitrygx WDYT?

if I'm not mistaken @bureddy faced the problem with hashing when removed ZEROing memory pointed by `peer_addr`.

Reply:
this was a different issue - conn hash was initialized with a wrong address length

Reply:
but actually, it shouldn't be used by Conn-matching prior to setting a real address.
maybe ZEROing hides some real issue

Reply:
we fixed that issue where we did memset only part of the sockaddr_storage based on address size and use sizeof(sockaddr_storage) for hashing.  that issue is fixed  by dynamic sockaddr in the EP

Reply:
I will remove it. lets see if CI finds any issues